README.md

```diff
@@ -259,6 +259,42 @@ OrderState.values # ['CREATED', 'PAID']
 ShippedOrderState.values # ['CREATED', 'PAID', 'PREPARED', SHIPPED']
 ```
 
+### Exhaustive case matcher
+
+If you want to make sure that you cover all cases in a case stament, you can use the exhaustive case matcher: `Ruby::Enum::Ecase`. It will raise an error if a case/enum value is not handled or if a value is specified that's not part of the enum. This is inspired by the [Rust Pattern Syntax](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html).
+
+> NOTE: This will add checks at runtime which might lead to slightly worse performance.
+
+> NOTE: `:else` is a reserved keyword if you want to use `Ruby::Enum::Ecase`.
+
+```ruby
+class ShippedOrderState < OrderState
+  include Ruby::Enum::Ecase
+
+  define :PREPARED, 'PREPARED'
+  define :SHIPPED, 'SHIPPED'
+end
+```
+
+```ruby
+shipped_order_state = ShippedOrderState::SHIPPED
+ShippedOrderState.ecase(shipped_order_state, {
```

Comment:
What does this return?

Reply:
The return value of the block or an array of return values if multiple blocks matched. Added it to the README.